node.c

```diff
@@ -483,7 +511,7 @@ rb_ast_delete_mark_object(rb_ast_t *ast, VALUE obj)
 VALUE
 rb_ast_tokens(rb_ast_t *ast)
 {
-    return ast->node_buffer->tokens;
+    return (VALUE)ast->node_buffer->tokens;
```

Comment:
Is `rb_parser_ary_t` compatible with `VALUE`?
This function is used in the implementation of `RubyVM::AbstractSyntaxTree::Node#all_tokens`, and the return value will be seen in Ruby level.

Reply:
I reimplemented `rb_ast_tokens()` that constructs RArray from rb_parser_ary_t https://github.com/ruby/ruby/commit/8a986a2ba1e7956ce804c77032e8d4da1e6357db